semantic-python/src/Language/Python/Core.hs

```diff
@@ -39,9 +53,10 @@ type CoreSyntax sig t = ( Member Core sig
                         )
 
 class Compile py where
-  -- FIXME: we should really try not to fail
```

Comment:
IMO this `FIXME` still applies; it’d be nice to eliminate the `MonadFail` constraint.

Reply:
I’m not sure we can eliminate it. We want always to fail when we hit something like an `eval` statement, right?

Reply:
Nah, failing is way too coarse-grained for that. We want to emit core that throws, sure, but not fail the compilation outright.

Reply:
Core that throws you say… intriguing. Okay, I’ll put this back in

Reply:
Yep. That way, `eval` in unreachable code poses no problem; and we can compute the analysis modulo any failures from `eval` the same way as we can compute it modulo any failures from division by zero.